src/gcrypt/crypto.c

```diff
@@ -17,18 +17,13 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
-#include "../system.h"
-
-#include <gcrypt.h>
-
+#include "../random.h"
 #include "../crypto.h"
 
 void crypto_init(void) {
+	random_init();
```

Comment:
I think the crypto functions shouldn't call `random_init()` and `random_exit()` anymore then, as that's still a bit of duplication.

Reply:
Calls to `crypto_exit()` were missing in some places, so I had to rearrange some code to avoid repeating `random_exit` before two dozen `return`s.